src/modules/regions/regions.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Regions } from './entities/region.entity';
+import { ApiOperation, ApiResponse } from '@nestjs/swagger';
+
+@Injectable()
+export class RegionsService {
+  constructor(
+    @InjectRepository(Regions)
+    private regionsRepository: Repository<Regions>
+  ) {}
+
+  @ApiOperation({ summary: 'Get all regions' })
+  @ApiResponse({
+    status: 200,
+    description: 'The regions have been successfully fetched.',
+    schema: {
+      example: {
+        status: 'success',
+        data: {
+          regions: [
+            {
+              id: 'string',
+              region: 'String',
+            },
+          ],
+        },
+      },
+    },
+  })
+  @ApiResponse({ status: 401, description: 'Unauthorized.' })
+  async findAll(): Promise<Regions[]> {
```

Comment:
make your method name more descriptive. something like getAllRegions is fine

Reply:
@Homoakin619  i've made the changes